sklearn/utils/validation.py

```diff
@@ -43,12 +43,16 @@ def _assert_all_finite(X):
 def assert_all_finite(X):
     """Throw a ValueError if X contains NaN or infinity.
 
-    Input MUST be an np.ndarray instance or a scipy.sparse matrix."""
+    Parameters
+    ----------
+    X : np.ndarray, scipy.sparse
+        ``X`` MUST be an np.ndarray instance or a scipy.sparse matrix.
```

Comment:
I would remove this line. `array-like` is what we tend to use when we want to say that the input can be a list